### ISSUE TITLE
[member-naming] @internal

### ISSUE THREAD

**Issue by htho** [typescript-eslint/typescript-eslint#1221]
**Repro**
```JSON
{
  "rules": {
    "@typescript-eslint/member-naming": ["error", {
      "private": "^_",
      "protected": "^_",
      "@internal": "^_"
    }]
  }
}
```

```TS
class Test {
  private a; // -> error
  private _a; // -> ok

  protected b; // -> error
  protected _b; // -> ok

  /** @internal */
  c; // -> error
  /** @internal */
  _c; // -> ok

  public d; // -> ok
}
```

**Expected Result**
```TS
class Test {
  /** @internal */
  c; // -> error
  /** @internal */
  _c; // -> ok
}
```
**Actual Result**
```TS
class Test {
  /** @internal */
  c; // -> ok (implicitly public)
  /** @internal */
  _c; // -> ok (implicitly public)
}
```
**Additional Info**
As mentioned in #1215, `@internal` should be a common modifier. Therefore it makes sense to add rules for internal properties.

**Comment by bradzacher** [comment]
> `@internal` should be a common modifier

it's not that common. most projects don't emit publicly consumed type defs, so most people don't use it, nor do they know that it exists. The fact that this rule has been in for a few years and this is the first request we've gotten for this is evidence enough of this.

Also, be careful on github. @ < word > will tag someone. In this case you've tagged some poor sod who is now going to receive an email for your tag. Wrapping things in backticks converts it to a code fragment so github won't make it into a tag.

---

As with #1215, this is no small task. Happy to accept a PR if you want to put one up.

**Comment by htho** [comment]
> Also, be careful on github. @ < word > will tag someone. In this case you've tagged some poor sod who is now going to receive an email for your tag. Wrapping things in backticks converts it to a code fragment so github won't make it into a tag.

You are right, I forgot about that. But hey: If you chose a generic username, you should expect to be bothered with off-topic content from time to time. Imagine someone would actually have a mailbox `@example.com` ;-)

**Comment by JoshuaKGoldberg** [comment]
Ah, this was missed in #8792 because of the rename.
